tests/python_functional/src/syslog_ng_config/statements/sources/network_source.py

```diff
@@ -20,11 +20,29 @@
 # COPYING for details.
 #
 #############################################################################
+from src.driver_io.network.network_io import NetworkIO
 from src.syslog_ng_config.statements.sources.source_driver import SourceDriver
 
 
 class NetworkSource(SourceDriver):
     def __init__(self, **options):
         self.driver_name = "network"
-        super(NetworkSource, self).__init__(options=options)
-        self.source_writer = None  # Will be added later, using Loggen
+        self.io = self.__create_io(options)
+        super(NetworkSource, self).__init__(logger_name=__name__, options=options)
+
+    def write_log(self, formatted_content, rate=None):
+        self.io.write(formatted_content, rate=rate)
+
+        super(NetworkSource, self).write_log(
```

Comment:
This call looks like as if it was part of the transport logic. But, if i understood correctly, this is used just to write a diagnostical log. The transportation actually happens in the previous line.

It is confusing for me that we decouple a diagnostical log from its action. Even into a different class.

I most definitely did not understand the problem to it's core, but is there any alternative? To workaround the mismatch of `__name__`? Can we pass a logger instance around instead? Or access it from "super"? Or create a "log" method in the superclass?